calendar-bundle/src/Resources/contao/modules/ModuleEventReader.php

```diff
@@ -392,6 +399,24 @@ protected function compile()
 
 		$this->Template->event = $objTemplate->parse();
 
+		// Add cache headers for content elements in the future if they have not been generated.
+		if ($needsCacheHeaders)
+		{
+			$objElement = ContentModel::findUpcomingByPidAndTable($id, 'tl_calendar_events');
```

Comment:
Imho it's enough fetch the first upcoming element (ordered by `start` date) to define the max age.